src/app/view/components/dot-navigation/dot-navigation.component.ts

```diff
@@ -1,21 +1,40 @@
-import { Component, OnInit } from '@angular/core';
-import { Observable } from 'rxjs/Observable';
-import { DotMenu } from '../../../shared/models/navigation';
-import { DotNavigationService } from './dot-navigation.service';
+import { Component, OnInit, Input, OnChanges, SimpleChanges, EventEmitter, Output } from '@angular/core';
+import { take } from 'rxjs/operators';
+import { NavigationEnd } from '@angular/router';
+
+import { DotMenu, DotMenuItem } from '../../../shared/models/navigation';
+import { DotNavigationService } from './services/dot-navigation.service';
 
 @Component({
     providers: [],
     selector: 'dot-main-nav',
     styleUrls: ['./dot-navigation.component.scss'],
     templateUrl: 'dot-navigation.component.html'
 })
-export class DotNavigationComponent implements OnInit {
-    menu: Observable<DotMenu[]>;
+export class DotNavigationComponent implements OnInit, OnChanges {
+    @Input() collapsed = false;
+    @Output() change = new EventEmitter<boolean>();
+    menu: DotMenu[];
 
     constructor(private dotNavigationService: DotNavigationService) {}
 
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.collapsed.currentValue) {
+            this.menu = this.menu.map((item: DotMenu) => {
+                item.isOpen = false;
+                return item;
+            });
+        }
+    }
+
     ngOnInit() {
-        this.menu = this.dotNavigationService.items$;
+        this.dotNavigationService.items$.pipe(take(1)).subscribe((menu: DotMenu[]) => {
+            this.menu = menu;
+        });
+
+        this.dotNavigationService.onNavigationEnd().subscribe((event: NavigationEnd) => {
+            this.setActive(event.url.split('/').pop());
```

Comment:
I think it is going to fail with url like:

- http://localhost:8080/dotAdmin/#/content-types-angular/edit/c541abb1-69b3-4bc5-8430-5e09e5239cc8
- http://localhost:8080/dotAdmin/#/content-types-angular/create/content